### PR TITLE
chore: (analytics) Report which UT frontend package is being used to make requests

### PR DESCRIPTION
**Pull request by markflorkowski**

### CODE REVIEW

File: packages/react/src/useUploadThing.ts [hunk 1]
```diff
@@ -103,6 +103,7 @@ export const INTERNAL_uploadthingHookGen = <
             opts.onUploadBegin(file);
           },
           url: initOpts.url,
+          package: "uploadthing-react",
```
Comment (juliusmarminge) [inline packages/react/src/useUploadThing.ts:106]:
```suggestion
          package: "@uploadthing/react",
```

File: packages/react/src/useUploadThing.ts [hunk 2]
```diff
@@ -164,6 +165,7 @@ export const generateReactHelpers = <TRouter extends FileRouter>(initOpts?: {
       DANGEROUS__uploadFiles<TRouter, TEndpoint>(endpoint, {
         ...opts,
         url,
+        package: "uploadthing-react",
```
Comment (juliusmarminge) [inline packages/react/src/useUploadThing.ts:168]:
```suggestion
        package: "@uploadthing/react",
```

File: packages/uploadthing/src/client.ts [hunk 3]
```diff
@@ -45,6 +45,14 @@ type UploadFilesOptions<
    * @example URL { https://www.example.com/api/uploadthing }
    */
   url: URL;
+
+  /**
+   * The uploadthing package that is making this request
+   * @example "uploadthing-react"
```
Comment (juliusmarminge) [inline packages/uploadthing/src/client.ts:51]:
```suggestion
   * @example "@uploadthing/react"
```

File: packages/uploadthing/src/client.ts [hunk 4]
```diff
@@ -243,10 +253,20 @@ export const genUploader = <TRouter extends FileRouter>(initOpts: {
    * @default (VERCEL_URL ?? window.location.origin) + "/api/uploadthing"
    */
   url?: string | URL;
+
+  /**
+   * The uploadthing package that is making this request
+   * @example "uploadthing-react"
```
Comment (juliusmarminge) [inline packages/uploadthing/src/client.ts:259]:
```suggestion
   * @example "@uploadthing/react"
```

File: packages/uploadthing/src/internal/handler.ts [hunk 5]
```diff
@@ -151,6 +152,7 @@ export const buildRequestHandler = <TRouter extends FileRouter>(
     const uploadthingHook = getHeader(req, "uploadthing-hook") ?? undefined;
     const slug = params.get("slug") ?? undefined;
     const actionType = (params.get("actionType") as ActionType) ?? undefined;
+    const utPackage = getHeader(req, "x-ut-package") ?? "unknown";
```
Comment (juliusmarminge) [inline packages/uploadthing/src/internal/handler.ts:155]: why is this `x-ut-package` when its `x-uploadthing-package` above?

Comment (markflorkowski) [reply packages/uploadthing/src/internal/handler.ts:155]: because I am not good at writing code.

**Review by juliusmarminge — APPROVED** [review]
Should we also track what backend adapter is used?

**Comment by markflorkowski** [comment]
> Should we also track what backend adapter is used?

Probably not a bad idea

**Pull request merged**